include/veriblock/blockchain/block_index.hpp

```diff
@@ -167,6 +169,11 @@ struct BlockIndex {
     return nullptr;
   }
 
+  bool isExistingPid(const pid_t& pid) const {
```

Comment:
more efficient would be to create a set of ids from this vector, and then try to insert into that set, and if item is inserted, then there's no duplicate.

```
std::set<pid_t> pids;
if(pids.insert(pid).second) {
  // can be inserted, unique element
} else {
  // duplicate
}
```
```

Reply:
Why do you think it is more efficient? I see that constructing set from vector takes O(NlogN) and vector lookup is O(N).